### PR TITLE
Support for adding values to readonly fields, while creation

### PR DESCRIPTION
**Pull request by jagadeeshym**

### CODE REVIEW

File: src/components/IterableEditComponent.tsx [hunk 1]
```diff
@@ -241,7 +245,7 @@ export class IterableEditComponent extends React.Component<ImageUploadProps | It
                                 <div style={this.props.field.iterabletype.style ?
                                     Object.assign({}, this.props.field.iterabletype.style, { display: "inline-block" }) : { display: "inline-block" }}>
                                     <MultiSelectComponent
-                                        readonly={getReadOnly(this.props.field.iterabletype.readonly, currentModel) || this.props.readonly}
+                                        readonly={this.checkReadonly(this.props.field.iterabletype.readonly, currentModel)}
```
Comment (kunal15595) [inline src/components/IterableEditComponent.tsx]: this is repeated many times. pull it out.

File: src/components/IterableEditComponent.tsx [hunk 2]
```diff
@@ -547,7 +551,7 @@ export class IterableEditComponent extends React.Component<ImageUploadProps | It
                                 onMouseEnter={this.showIterableButtons.bind(this, index)}
                                 onMouseLeave={this.hideIterableButtons.bind(this, index)}>
                                 <input
-                                    disabled={getReadOnly(this.props.field.iterabletype.readonly, datum) || this.props.readonly}
+                                    disabled={this.checkReadonly(this.props.field.iterabletype.readonly, datum)}
```
Comment (kunal15595) [inline src/components/IterableEditComponent.tsx]: I think datum is same as currentModel, use currentModel only.

File: src/components/NestedEditComponent.tsx [hunk 3]
```diff
@@ -198,7 +202,7 @@ export class NestedEditComponent extends React.Component<InlineComponentProps, a
             const currentModel = (this.props.currentModel !== undefined && this.props.currentModel[field.field] !== undefined) ? this.props.currentModel[field.field] : {}
             return (
                 <CheckboxComponent field={field}
-                    readonly={getReadOnly(field.readonly, currentModel) || this.props.readonly}
+                    readonly={this.checkReadonly(field.readonly, currentModel)}
```
Comment (kunal15595) [inline src/components/NestedEditComponent.tsx]: pull out.

File: src/components/IterableEditComponent.tsx (hunk 2 above)
Comment (jagadeeshym) [reply src/components/IterableEditComponent.tsx]: Updated

File: src/components/IterableEditComponent.tsx (hunk 1 above)
Comment (jagadeeshym) [reply src/components/IterableEditComponent.tsx]: Updated

**Pull request merged**